### PR TITLE
Invitation and Organization model service cleanup

### PR DESCRIPTION
**Pull request by alexeykazakov**
- [x] Remove Organization Controller Service (use Org Model Service directly from Controller)
- [x] Add Org model service to application
- [x] Create Org model service by using application interface (fixes Org transaction)
- [x] Start using local unexported repository interfaces
- [x] Package names cleanup (in imports)

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-453-3`

File: authorization/invitation/model/invitation_model_service.go [hunk 1]
```diff
@@ -2,41 +2,51 @@ package model
...
 	"github.com/fabric8-services/fabric8-auth/authorization"
 	"github.com/fabric8-services/fabric8-auth/authorization/invitation"
-	invRepo "github.com/fabric8-services/fabric8-auth/authorization/invitation/repository"
-	permissionService "github.com/fabric8-services/fabric8-auth/authorization/permission/model"
-	"github.com/fabric8-services/fabric8-auth/authorization/repository"
-	resourceRepo "github.com/fabric8-services/fabric8-auth/authorization/resource/repository"
+	invitationrepo "github.com/fabric8-services/fabric8-auth/authorization/invitation/repository"
+	permissionsrv "github.com/fabric8-services/fabric8-auth/authorization/permission/model"
+	resource "github.com/fabric8-services/fabric8-auth/authorization/resource/repository"
+	resourcerepo "github.com/fabric8-services/fabric8-auth/authorization/resource/repository"
```
Comment (sbryzak) [inline authorization/invitation/model/invitation_model_service.go]: This is being imported twice

File: authorization/invitation/model/invitation_model_service.go [hunk 2]
```diff
@@ -2,41 +2,51 @@ package model
...
+	role "github.com/fabric8-services/fabric8-auth/authorization/role/repository"
 	"github.com/fabric8-services/fabric8-auth/errors"
+
 	"github.com/satori/go.uuid"
 )
 
+// repositories interface lets us avoid an import cycle
+type repositories interface {
```
Comment (sbryzak) [inline authorization/invitation/model/invitation_model_service.go]: Nice idea, I identified this as an alternative if import cycles became a serious problem.

File: authorization/invitation/model/invitation_model_service.go (hunk 1 above)
Comment (alexeykazakov) [reply authorization/invitation/model/invitation_model_service.go]: Fixed - https://github.com/fabric8-services/fabric8-auth/pull/453/commits/e1719ad7f74cc2056a1f0f689203fe9e68cd50f2

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-453-4`

**Pull request merged**